homeassistant/components/modbus/modbus.py

```diff
@@ -311,6 +311,10 @@ async def async_pb_connect(self) -> None:
         async with self._lock:
             try:
                 await self._client.connect()  # type: ignore[union-attr]
+                if not self._client or not self._client.connected:
```

Comment:
Are you sure that those lines (314-317) of code are really useful and called?
As you can see, modbus raises an exception in case of connection failure and it retries..

Reply:
The code never throws an exception for me. When you review the logs from the attached issue https://github.com/home-assistant/core/issues/129940 you can see that pymodbus comes back with Errno 111, however the core modbus.py assumes the connection was established and tries communicating.
Lines 314-317 detects that.